### DISCUSSION TITLE
How to display prop cardinality?

### DISCUSSION THREAD

**Discussion by VladimirAlexiev** [VladimirAlexiev/rdf2rml#43] [Q&A]
Einar Clementz asked
> In the To-Do part of your description, you talk about cardinality support. Is there any support for some cardinality constructs from RDF to puml as of now? One of the reasons to go to such tooling that rdfpuml is, is to get visual support for cardinalities. I deal with classic E/R people, cardinalities visual is key enabler.

**Comment by VladimirAlexiev** [comment]
For now it's just an idea.
There are a couple of variants described, 
- The more viable one is https://github.com/VladimirAlexiev/rdf2rml#cardinality-with-blank-node
- (I don't know which perl library supports RDF-star)

The standard ways to describe cardinality involve:
- class and restriction, eg
```ttl
:x a :Class, :prop :y.
:Class a owl:Class;
  rdfs:subClassOf [a owl:Restriction; owl:onProperty :prop; owl:minCardinality m; owl:maxCardinality n]
  # or owl:minQualifiedCardinality m; owl:maxQualifiedCardinality
```
- class and node shape, eg
```ttl
:x a :Class, :prop :y.
:Class a owl:Class.
:Shape a sh:NodeShape, sh:targetClass :Class.
  # or :Class a owl:Class; sh:NodeShape.
:Shape sh:property [sh:path :prop; sh:minCount m; sh:maxCount n].
```

But I use the presence of statements about classes as a signal to display the class as a node... which users may not want...
Basically, the tool was designed to display instances, but people also want to display ontology info...

If this feature is important for you, please use "Create issue from discussion" and then add some details as to how you see it expressed in turtle.
